diff --git a/pn8x/halimpl/hal/phNxpNciHal_ext.cc b/pn8x/halimpl/hal/phNxpNciHal_ext.cc
index 9fb21b6..9ca42fc 100644
--- a/pn8x/halimpl/hal/phNxpNciHal_ext.cc
+++ b/pn8x/halimpl/hal/phNxpNciHal_ext.cc
@@ -766,9 +766,11 @@ NFCSTATUS phNxpNciHal_write_ext(uint16_t* cmd_len, uint8_t* p_cmd_data,
          p_cmd_data[4] == 0x24 || p_cmd_data[4] == 0x27 ||
          p_cmd_data[4] == 0x28 || p_cmd_data[4] == 0x29 ||
          p_cmd_data[4] == 0x2a)) {
-      NXPLOG_NCIHAL_D("> Send EOF set");
-      icode_send_eof = 1;
+        if (nxpncihal_ctrl.nci_info.nci_version != NCI_VERSION_2_0) {
+            NXPLOG_NCIHAL_D("> Going through workaround - Send EOF set");
+            icode_send_eof = 1;
     }
+  }
 
     if (p_cmd_data[3] == 0x20 || p_cmd_data[3] == 0x24 ||
         p_cmd_data[3] == 0x60) {
